.github/workflows/main.yml

```diff
@@ -1,79 +1,256 @@
+          name: conda-env-file
+          path: environment.yaml
+          if-no-files-found: error
 
-  Tests_sql:
-    name: Tests SQL
+      - name: cleanup generated env files
+        run: rm environment.yaml
+
+  test_no_backends:
+    runs-on: ${{ matrix.os }}
+    needs:
+      - gen_conda_env_file
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python-version:
+          - "3.7"
+          - "3.9"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: download env file
+        uses: actions/download-artifact@v2
+        with:
+          name: conda-env-file
+          path: tmp
+
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          mamba-version: "*"
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          channel-priority: strict
+          activate-environment: ibis
+          python-version: ${{ matrix.python-version }}
+          environment-file: tmp/environment.yaml
+          condarc-file: ci/condarc
+
+      - name: install ibis
+        run: pip install .
+
+      - name: run tests
+        run: |
+          pytest \
+            ibis/tests \
+            --numprocesses auto \
+            --quiet \
+            -ra \
+            --junitxml=junit.xml \
+            --cov=ibis \
+            --cov-report=xml:coverage.xml | tee pytest.log
+
+      - name: pytest errors
+        run: ./ci/pytest_errors.sh
+        if: failure()
+
+  test_simple_backends:
+    runs-on: ${{ matrix.os }}
+    needs:
+      - gen_conda_env_file
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python-version:
+          - "3.7"
+          - "3.9"
+        pyarrow:
+          - "1"
+          - "5"
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: download env file
+        uses: actions/download-artifact@v2
+        with:
+          name: conda-env-file
+          path: tmp
+
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          mamba-version: "*"
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          channel-priority: strict
+          activate-environment: ibis
+          python-version: ${{ matrix.python-version }}
+          environment-file: tmp/environment.yaml
+          condarc-file: ci/condarc
+
+      - name: install pyarrow
+        run: conda install -n ibis pyarrow=${{ matrix.pyarrow }}
+
+      - name: install ibis
+        run: pip install .
+
+      - name: download backend data
+        run: python ci/datamgr.py download
+
+      - name: set backends
+        id: set_backends
+        run: echo '::set-output name=backends::["csv", "dask", "hdf5", "pandas", "parquet", "sqlite"]'
```

Comment:
Could we just hardcode the backends and call the appropiate `datamgr.py` commands? 

```bash
python ci/datamgr.py download
python ci/datamgr.py parquet
python ci/datamgr.py sqlite
# the rest of the click commands do nothing according to the source code
```

Reply:
Roger that. Will fix.

Reply:
Not necessary after matrix-ing the jobs.